proofs/tree.go

```diff
@@ -516,9 +517,20 @@ func (s LeafList) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
 
-// Less compares two strings lexicographically
-func (s LeafList) Less(i, j int) bool {
-	return strings.Compare(string(s[i].Property.ReadableName()), string(s[j].Property.ReadableName())) == -1
+type sortByReadableName struct{ LeafList }
+// Compare by property name lexicographically
+func (m sortByReadableName) Less(i, j int) bool {
+	return strings.Compare(string(m.LeafList[i].Property.ReadableName()), string(m.LeafList[j].Property.ReadableName())) == -1
+}
+
+type sortByCompactName struct{ LeafList }
+// Compare by property conpact name 
+func (m sortByCompactName) Less(i, j int) bool {
+	first := new(bytes.Buffer)
```

Comment:
It would be nice to have a method on the Property to get the `[]byte` value and use this method here as well as here: https://github.com/centrifuge/precise-proofs/blob/master/proofs/property.go#L162

Reply:
You are right
func (m sortByCompactName) Less(i, j int) bool will be change to 
{
return bytes.Compare(AsBytes(m.LeafList[i].Property.Name(true)),AsBytes(m.LeafList[j].Property.Name(true))) ==-1
}